### ISSUE TITLE
SQLite driver marks `AUTOINCREMENT` or `ROWID` columns as generated

### ISSUE THREAD

**Issue by burdiyan** [stephenafamo/bob#44]
At least with SQLite, when a table has primary key and foreign key constraints defined on the same field, the code generation doesn't generate the field on the model (I guess because it thinks it's a relation).

An example schema:

```
CREATE TABLE a (id INTEGER PRIMARY KEY);

CREATE TABLE b (
    id INTEGER PRIMARY KEY,
    some_other_field TEXT,
    FOREIGN KEY (id) REFERENCES a (id)
);
```

**Comment by stephenafamo** [comment]
No relation fields are generated directly on the model. The relation should be attached to the `R` field which would be of the type `modelR` (dependendent on the name of the model).

**Comment by stephenafamo** [comment]
Is there no `R` field on the struct? If there is, what does it look like?

**Comment by burdiyan** [comment]
No, the `R` field looks fine (I guess, still learning this part of bob). But the problem is that the ID *field* doesn't get generated on the model, and looks like other parts of code are expecting it and things don't compile.

**Comment by stephenafamo** [comment]
Fixed and released in `v0.20.1`

**Comment by stephenafamo** [comment]
Fixed and released in `v0.20.1`

**Comment by burdiyan** [comment]
Thanks for the quick response! So far, I'm pretty impressed with the quality of documentation and good reasoning behind how things are put together, congrats!
